### PR TITLE
SNOW-1825719 - setting proxy port if default was skipped during parsing

### PR DESCRIPTION
**Pull request by sfc-gh-pmotacki**
### Description
Please explain the changes you made here.

### Checklist
- [ ] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [ ] Create tests which fail without the change (if possible)
- [ ] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [ ] Extend the README / documentation and ensure is properly displayed (if necessary)
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

**Review by sfc-gh-dprzybysz — APPROVED** [review]

File: lib/util.js [hunk 1]
```diff
@@ -872,11 +879,13 @@ exports.getProxyFromEnv = function (isHttps = true) {
...
     Logger.getInstance().info('Util.getProxyEnv: the protocol was missing from the environment proxy. Use the HTTP protocol.');
     proxyFromEnv = 'http' + '://' + proxyFromEnv;
+    isHttps = false;
   }
   proxyFromEnv = new URL(proxyFromEnv);
+  const port = getDefaultPortIfNotSet(proxyFromEnv, isHttps);
   const proxy = {
     host: this.validateEmptyString(proxyFromEnv.hostname),
-    port: Number(this.validateEmptyString(proxyFromEnv.port)),
+    port: Number(this.validateEmptyString(port)),
```
Comment (sfc-gh-astachowski) [inline lib/util.js:888]: I think this validate is no longer needed

Comment (sfc-gh-pmotacki) [reply lib/util.js]: Ok, fixed

File: lib/util.js [hunk 2]
```diff
@@ -871,12 +878,13 @@ exports.getProxyFromEnv = function (isHttps = true) {
   Logger.getInstance().debug(`Util.getProxyEnv: Using ${protocol.toUpperCase()}_PROXY from the environment variable`);
   if (proxyFromEnv.indexOf('://') === -1) {
     Logger.getInstance().info('Util.getProxyEnv: the protocol was missing from the environment proxy. Use the HTTP protocol.');
-    proxyFromEnv = 'http' + '://' + proxyFromEnv;
+    proxyFromEnv = protocol + '://' + proxyFromEnv;
```
Comment (sfc-gh-astachowski) [inline lib/util.js:881]: shouldn't we check if we need to change isHttps here?

File: lib/util.js [hunk 3]
```diff
@@ -871,12 +878,13 @@ exports.getProxyFromEnv = function (isHttps = true) {
   Logger.getInstance().debug(`Util.getProxyEnv: Using ${protocol.toUpperCase()}_PROXY from the environment variable`);
   if (proxyFromEnv.indexOf('://') === -1) {
     Logger.getInstance().info('Util.getProxyEnv: the protocol was missing from the environment proxy. Use the HTTP protocol.');
```
Comment (sfc-gh-astachowski) [inline lib/util.js:880]: We should change this to log proper protocol

Comment (sfc-gh-dszmolka) [reply lib/util.js:880]: this line is not part of this change; but the earlier implementation with #938 . 
It was agreed, that we follow the practice set in the industry and the other drivers, and will use the HTTP protocol if the user does not set it in the envvar.

File: lib/util.js (hunk 2 above)
Comment (sfc-gh-dszmolka) [inline lib/util.js:881]: please, let's use HTTP if the user doesn't define the protocol. See above comment.

**Review by sfc-gh-dszmolka — CHANGES_REQUESTED** [review]

**Review by sfc-gh-dheyman-1 — APPROVED** [review]

**Pull request merged**